src/cmd-push-container-manifest

```diff
@@ -58,26 +85,25 @@ def main():
                 raise Exception
             images.append(f"oci-archive:{ociarchive}")
 
-        # Create/Upload the manifest list to the container registry
-        manifest_info = create_and_push_container_manifest(
-            args.repo, args.tags, images, args.v2s2)
-
-        # Update the `meta.json` files. Note the digest included is the
-        # arch-specific one for each individual arch, and not the manifest list
-        # digest. See: https://github.com/coreos/coreos-assembler/issues/3122.
-        assert len(manifest_info['manifests']) == len(buildmetas)
-        for manifest in manifest_info['manifests']:
-            arch = manifest['platform']['architecture']
-            if arch == 'arm64':
-                arch = 'aarch64'
-            elif arch == 'amd64':
-                arch = 'x86_64'
-            buildmetas[arch][args.metajsonname] = {
-                'image': args.repo,
-                'digest': manifest['digest'],
-                'tags': args.tags
-            }
-            buildmetas[arch].write(artifact_name=args.metajsonname)
+        if upload is True or args.force is True:
```

Comment:
I'm thinking that in the case we decide there is nothing to do we should output and log message to that effect. In the same vein it might make sense to check that here and print the log message and return early.

```suggestion
        if not upload and not args.force:
            print(f"Detected manifest list in remote registry already matches desired state. Skipping manifest list push.")
            return
```

You can probably come up with a better message than that, but I think that illustrates the idea.

Reply:
That also has the nice property of avoiding another indent level. :)